src/torrent.cpp

```diff
@@ -302,7 +302,10 @@ namespace libtorrent
 			e.fail_limit = 0;
 			e.source = announce_entry::source_magnet_link;
 			e.tier = tier;
-			m_trackers.push_back(e);
+			if (!find_tracker(e.url))
```

Comment:
as you point out ``add_tracker`` has side effects not desirable here, so I think we'd need to stick with just push_back